src/Xamarin.Android.Build.Tasks/Tests/Xamarin.ProjectTools/Android/AndroidSdkResolver.cs

```diff
@@ -131,19 +138,41 @@ static int GetMaxInstalledPlatformInternal ()
 			if (maxInstalled != null)
 				return maxInstalled.Value;
 
+			var supportedVersions = new AndroidVersions (GetAndroidApiInfoDirectories ());
+
 			string sdkPath = GetAndroidSdkPath ();
 			foreach (var dir in Directory.EnumerateDirectories (Path.Combine (sdkPath, "platforms"))) {
 				int version;
 				string v = Path.GetFileName (dir).Replace ("android-", "");
 				Console.WriteLine ($"GetMaxInstalledPlatform: Parsing {v}");
 				if (!int.TryParse (v, out version))
 					continue;
-				if (version < maxInstalled)
+				if (version < maxInstalled || version > supportedVersions.MaxStableVersion?.ApiLevel)
```

Comment:
It seemed like it was tricky here to get the max supported API level.

Maybe we could somehow read this value instead?

https://github.com/xamarin/xamarin-android/blob/9bbda819d68d982a76145506690ed936b3433c11/Configuration.props#L33

Maybe `xaprepare` could put this value as a `const int` in this file?

https://github.com/xamarin/xamarin-android/blob/9bbda819d68d982a76145506690ed936b3433c11/build-tools/xaprepare/xaprepare/Steps/Step_GenerateFiles.cs#L155-L169

So you'd just end up with:

```suggestion
				if (version < maxInstalled || version > XABuildConfig.AndroidLatestStableApiLevel)
```

Reply:
This seems like a good idea

Reply:
Fair point that this approach should be less error prone, updated.